common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2TokenCache.java

```diff
@@ -52,11 +54,28 @@ public OAuth2TokenCache(Context context) {
      * @param oAuth2Strategy The strategy used to create the token request.
      * @param request        The request used to acquire tokens and credentials.
      * @param response       The response received from the IdP/STS.
+     * @return The {@link ISaveTokenResult} containing the Account + Credentials saved to the cache.
      * @throws ClientException If tokens cannot be successfully saved.
      */
-    public abstract void saveTokens(final T oAuth2Strategy,
-                                    final U request,
-                                    final V response) throws ClientException;
+    public abstract ISaveTokenResult saveTokens(final T oAuth2Strategy,
+                                                final U request,
+                                                final V response) throws ClientException;
+
+    /**
+     * Loads the tokens for the supplied Account into the result {@link ISaveTokenResult}.
+     *
+     * @param account The Account whose Credentials should be loaded.
+     * @return The resulting ISaveTokenResult. Entries may be empty if not present in the cache.
+     */
+    public abstract ISaveTokenResult loadTokens(final Account account);
```

Comment:
This method's signature is likely going to change to also accept a `clientId` so that these transactions are app-specific.